packages/next-playwright/src/msw.ts

```diff
@@ -0,0 +1,111 @@
+import { test as base } from './index'
+import type { NextFixture } from './next-fixture'
+import {
+  type RequestHandler,
+  type MockedResponse,
+  MockedRequest,
+  handleRequest,
+} from 'msw'
+import { Emitter } from 'strict-event-emitter'
+
+export * from 'msw'
+export * from '@playwright/test'
+export type { NextFixture }
+
+export interface MswFixture {
+  use: (...handlers: RequestHandler[]) => void
+}
+
+export const test = base.extend<{
+  msw: MswFixture
+  mswHandlers: RequestHandler[]
+}>({
+  mswHandlers: [],
+
+  msw: [
+    async ({ next, mswHandlers }, use) => {
+      const handlers: RequestHandler[] = [...mswHandlers]
+      const emitter = new Emitter()
+
+      next.onFetch(async (request) => {
+        const {
+          body,
+          method,
+          headers,
+          credentials,
+          cache,
+          redirect,
+          integrity,
+          keepalive,
+          mode,
+          destination,
+          referrer,
+          referrerPolicy,
+        } = request
+        const mockedRequest = new MockedRequest(new URL(request.url), {
```

Comment:
You should be able to pass the `request` instance directly as the value of the second argument to `MockedRequest`. Afaik, the `RequestInit` and `Request` objects are compatible.